### PR TITLE
Fix `ln` file completion

### PR DESCRIPTION
**Pull request by gechr**
Fixes https://github.com/fish-shell/fish-shell/issues/4089

### CODE REVIEW

File: share/completions/ln.fish [hunk 1]
```diff
@@ -1,16 +1,16 @@
+complete -c ln -s s -l symbolic            -d 'Make symbolic links instead of hard links'
+complete -c ln      -l backup              -d 'Make a backup of each existing destination file' -a "none off numbered t existing nil simple never"
+complete -c ln -s b                        -d 'Make a backup of each existing destination file'
+complete -c ln -s dF -l directory          -d 'Allow superuser to attempt to hard link directories'
```
Comment (gechr) [inline share/completions/ln.fish]: FWIW I also included `-F` here as a bonus! 😄 

```
› ln --help | grep -w F
  -d, -F, --directory         allow the superuser to attempt to hard link
```

Comment (krader1961) [inline share/completions/ln.fish]: Note that more than a single letter short option is not documented to work. From the `complete` documentation:

> SHORT_OPTION is a one character option for the command.

And if you press \<tab\> after typing `-` then try to select that you get just `-d`. On the other hand if you type `-d` and press \<tab\> it adds the `F`. Not sure if both behaviors are intentional, accidental, or an error but we probably don't want to rely on it. If you want to support `-F` as an alias for `-s` you need a separate completion.

Comment (gechr) [inline share/completions/ln.fish]: Ahh! I think it's less hassle to just remove it. I doubt it's useful to anyone, just noticed it was missing and thought "why not" add it.

Commit updated with `-F` omitted.

**Comment by krader1961** [comment]
Merged as commt 5e39efa4. Thanks, in particular for leaving the `-f` flag on the version and help flags.

**Pull request closed without merging**